plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -453,49 +453,68 @@ public function postSave(PropelPDO $con = null)
 
 	public function addSchedulingData()
 	{
+		$language = '';
+		$feature = null;
+		/* @var $taskData kScheduledVendorTaskData */
+		$taskData = $this->getTaskJobData();
+		/* @var $connectedEvent LiveStreamScheduleEvent */
+		$connectedEvent = $taskData->getScheduleEvent();
+		/* @var $vendorCatalogItem VendorCatalogItem */
+		$vendorCatalogItem = VendorCatalogItemPeer::retrieveByPK($this->getCatalogItemId());
+
 		switch ($this->getServiceFeature())
 		{
+			case KalturaVendorServiceFeature::LIVE_TRANSLATION:
+			{
+				$language = languageCodeManager::getThreeCodeFromKalturaName($vendorCatalogItem->getTargetLanguage());
+				$this->addLanguageToFeature('LiveCaptionFeature', $connectedEvent, $taskData, $language);
+				break;
+			}
 			case KalturaVendorServiceFeature::LIVE_CAPTION:
-				/* @var $taskData kScheduledVendorTaskData */
-				$taskData = $this->getTaskJobData();
-				/* @var $connectedEvent LiveStreamScheduleEvent */
-				$connectedEvent = $taskData->getScheduleEvent();
-				/* @var $vendorCatalogItem VendorCatalogItem */
-				$vendorCatalogItem = VendorCatalogItemPeer::retrieveByPK($this->getCatalogItemId());
 				$language = languageCodeManager::getThreeCodeFromKalturaName($vendorCatalogItem->getSourceLanguage());
-
-				$feature = new LiveCaptionFeature();
-				$feature->setPreStartTime($connectedEvent->getStartDate(null) - $taskData->getStartDate());
-				$feature->setPostEndTime($taskData->getEndDate() - $connectedEvent->getEndDate(null));
-				$feature->setSystemName(LiveCaptionFeature::defaultName(LiveFeature::REACH_FEATURE_PREFIX . "-{$this->getId()}"));
-				$feature->setLanguage($language);
-
-				$connectedEvent->addFeature($feature, true);
-				$connectedEvent->save();
+				$this->addLanguageToFeature('LiveCaptionFeature', $connectedEvent, $taskData, $language);
 				break;
 		}
 	}
 
+	protected function addLanguageToFeature($featureType, &$connectedEvent, $taskData, $language,)
+	{
+		$feature = new $featureType();
+		$feature->setPreStartTime($connectedEvent->getStartDate(null) - $taskData->getStartDate());
+		$feature->setPostEndTime($taskData->getEndDate() - $connectedEvent->getEndDate(null));
+		$feature->setSystemName(LiveCaptionFeature::defaultName(LiveFeature::REACH_FEATURE_PREFIX . "-{$this->getId()}"));
```

Comment:
just like my comment in line 475, if you already explicitly call to live caption feature here there is no point providing the type from outside